covid19_sfbayarea/data/marin_scraper.py

```diff
@@ -0,0 +1,303 @@
+		# Each new row has data for a different gender.
+		split = row.split(',')
+		gender = split[0].lower()
+		if gender not in genders:
+			return ValueError('The genders have changed.')
+		c_gender[gender] = int(split[2])
+		d_gender[gender] = int(split[4])			
+
+	return c_gender, d_gender
+
+def get_breakdown_race_eth(chart_id: str, url: str) -> Tuple:
+	"""This method gets the breakdown of cases and deaths by race/ethnicity."""
+
+	csv_ = extract_csvs(chart_id, url)
+
+	csv_strs = csv_.splitlines()
+	keys = csv_strs[0].split(',') 
+	
+	if keys != ['Race/Ethnicity', 'COUNTY POPULATION', 'Case Count', 'Percent of Cases', 'Hospitalization Count', 'Percent of Hospitalizations', 'Death Count', 'Percent of Deaths']:
+		raise ValueError("The headers have changed.")
+
+	key_mapping = {"black/african american":"African_Amer", "hispanic/latino": "Latinx_or_Hispanic",
+            "american indian/alaska native": "Native_Amer", "native hawaiian/pacific islander": "Pacific_Islander", "white": "White", "asian": "Asian", "multi or other race": "Multi or Other Race"}
+            # "Multiple_Race", "Other" are not separate in this data set - they are one value under "Multi or Other Race"
+
+	c_race_eth = {}
+	d_race_eth = {}
+	
+	for row in csv_strs[1:]:
+		split = row.split(',')
+		race_eth = split[0].lower()
+		if race_eth not in key_mapping:
+			raise ValueError("The race_eth groups have changed.")
+		else:
+			c_race_eth[key_mapping[race_eth]] = int(split[2])
+			d_race_eth[key_mapping[race_eth]] = int(split[6])
+
+	return c_race_eth, d_race_eth
+
+def get_test_series(chart_id: str, url: str) -> Tuple:
+	"""This method gets the date, the number of positive and negative tests on that date, and the number of cumulative positive and negative tests."""
+
+	csv_ = extract_csvs(chart_id, url)
+	series = []
+
+	csv_strs = csv_.splitlines()
+	keys = csv_strs[0].split(',')
+	
+	test_history = []
+	
+	# Grab the dates, which are in the header
+	for entry in csv_strs[:1][0].split(',')[1:]: 
+		# need to exclude very first item in the csv_strs[:1][0].split(',') list (which is the value 'Date')
+		daily = {}
+		date_time_obj = datetime.strptime(entry, '%m/%d/%Y')
+		daily["date"] = date_time_obj.isoformat()
+		series.append(daily)
+
+	# The slicing makes this if statement hard to look at... there must be a better way?
+	if csv_strs[1:2][0].split(',')[:1][0] != 'Positive Tests' and csv_strs[2:][0].split(',')[:1][0] != 'Negative Tests':
+		raise ValueError('The kinds of tests have changed.')
+
+	# Grab the positive test result numbers, which is in the second row. 
+	# [1:] is included to make sure that 'Positive Tests' is not captured.
+	p_entries = csv_strs[1:2][0].split(',')[1:]
+	n_entries = csv_strs[2:][0].split(',')[1:]
+	
+	get_test_series_helper(series, p_entries, ['positive', 'cumul_pos'])
+	get_test_series_helper(series, n_entries, ['negative', 'cumul_neg'])
```

Comment:
To keep this whole thing simpler, it might be helpful to flip the columns and rows using [`zip`](https://docs.python.org/3.8/library/functions.html#zip):

```py
# Take several series (dates, positive cases, negative cases) and turn them
# into tuples of (date, positive, negative), one for each column.
dates, positives, negatives = [row.split(',')[:1] for row in csv_strs]
time_series = zip(dates, positives, negatives)

# Or more compact, though possibly tougher to read:
time_series = zip(*[row.split(',')[:1] for row in csv_strs])

# Results in:
[
  ('3/4/2020', 1, 1),
  ('3/5/2020', 0, 2),
  ('3/6/2020', 0, 9),
  ('3/7/2020', 0, 0),
  etc...
]
```

Reply:
Thanks for the tip Rob! I ended up taking your suggestion, and am trying to see if there's a way to do this that uses the csv module (right now I'm splitting by commas and like you've said, I should try to avoid that..)